### ISSUE TITLE
Unable to build 1.0.0-beta1 (QT errors)

### ISSUE THREAD

**Issue by Ravcii** [pgmodeler/pgmodeler#1714]
**Bug description**
Unable to build 1.0.0-beta1 version (beta branch at `0ab197e937a62b1d7fd43e574d5950e76577a620` commit). The only change compared to dev branch is version number so it's not working there too.

**How to reproduce**

1. `$ brew install qt libxml2 libpq`
2. Change pgmodeler.pri
<details>
<summary>pgmodeler.pri</summary>

```
macx {
  !defined(PGSQL_LIB, var): PGSQL_LIB = /opt/homebrew/Cellar/libpq/14.5/lib/libpq.dylib
  !defined(PGSQL_INC, var): PGSQL_INC = /opt/homebrew/Cellar/libpq/14.5/include
  !defined(XML_INC, var): XML_INC = /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libxml2/libxml
  !defined(XML_LIB, var): XML_LIB = /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libxml2.tbd
  INCLUDEPATH += "$$PGSQL_INC" "$$XML_INC"
}
```
</details>
I've tried brew's libxml2 before (just in case), but that didn't change anything.

3. Run `$ ./build.sh` (in the pgmodeler's root)
<details>
<summary>build.sh</summary>

```
export QT_ROOT=/opt/homebrew/Cellar/qt/6.3.1_4
export INSTALLATION_ROOT=~/PgModeler.app

$QT_ROOT/bin/qmake -r CONFIG+=release pgmodeler.pro \
    && make \
    && make install \
    && $QT_ROOT/bin/macdeployqt $INSTALLATION_ROOT \
                            $INSTALLATION_ROOT/Contents/MacOS/pgmodeler-ch \
                            $INSTALLATION_ROOT/Contents/MacOS/pgmodeler-cli
```
</details>

**Expected behavior**
PgModeler being built successfully

**Info about your desktop**
 - OS: macOS
 - Version: 12.6
 - Window manager:
 - pgModeler version: 1.0.0-beta1
 - Qt version: 6.3.1_4

**Stacktrace / Debug info**
<details>
<summary>errors</summary>

```

In file included from src/mainwindow.cpp:19:
In file included from src/mainwindow.h:43:
In file included from src/tools/databaseimportform.h:29:
src/tools/databaseimporthelper.h:51:3: error: unknown type name 'Catalog'
                Catalog catalog;
                ^
src/tools/databaseimporthelper.h:254:3: error: unknown type name 'Catalog'
                Catalog getCatalog();
                ^
In file included from src/mainwindow.cpp:19:
In file included from src/mainwindow.h:45:
In file included from src/tools/sqltoolwidget.h:36:
src/tools/databaseexplorerwidget.h:64:3: error: unknown type name 'Catalog'
                Catalog catalog;
                ^
In file included from src/mainwindow.cpp:19:
In file included from src/mainwindow.h:45:
In file included from src/tools/sqltoolwidget.h:37:
In file included from src/tools/sqlexecutionwidget.h:36:
src/utils/resultsetmodel.h:43:34: error: unknown type name 'Catalog'
                ResultSetModel(ResultSet &res, Catalog &catalog, QObject *parent = 0);
                                               ^
In file included from src/mainwindow.cpp:19:
In file included from src/mainwindow.h:45:
In file included from src/tools/sqltoolwidget.h:37:
src/tools/sqlexecutionwidget.h:116:32: error: unknown type name 'Catalog'
                static void fillResultsTable(Catalog &catalog, ResultSet &res, QTableWidget *results_tbw, bool store_data=false);
                                             ^
src/mainwindow.cpp:298:9: error: no matching member function for call to 'addWidget'
                grid->addWidget(sql_tool_wgt, 0, 0);
                ~~~~~~^~~~~~~~~
/opt/homebrew/lib/QtWidgets.framework/Headers/qgridlayout.h:101:10: note: candidate function not viable: no known conversion from 'SQLToolWidget *' to 'QWidget *' for 1st argument
    void addWidget(QWidget *, int row, int column, Qt::Alignment = Qt::Alignment());
         ^
/opt/homebrew/lib/QtWidgets.framework/Headers/qgridlayout.h:100:17: note: candidate function not viable: requires single argument 'w', but 3 arguments were provided
    inline void addWidget(QWidget *w) { QLayout::addWidget(w); }
                ^
/opt/homebrew/lib/QtWidgets.framework/Headers/qgridlayout.h:102:10: note: candidate function not viable: requires at least 5 arguments, but 3 were provided
    void addWidget(QWidget *, int row, int column, int rowSpan, int columnSpan, Qt::Alignment = Qt::Alignment());
         ^
src/mainwindow.cpp:838:14: error: comparison of distinct pointer types ('QObject *' and 'SQLToolWidget *')
                if(sender()!=sql_tool_wgt)
                   ~~~~~~~~^ ~~~~~~~~~~~~
In file included from src/mainwindow.cpp:19:
In file included from src/mainwindow.h:28:
In file included from /opt/homebrew/lib/QtWidgets.framework/Headers/QtWidgets:3:
In file included from /opt/homebrew/lib/QtWidgets.framework/Headers/QtWidgetsDepends:3:
In file included from /opt/homebrew/lib/QtCore.framework/Headers/QtCore:9:
In file included from /opt/homebrew/lib/QtCore.framework/Headers/qabstractanimation.h:43:
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:309:16: error: no matching function for call to 'connect'
        return connect(sender, signal, sender, std::move(slot), Qt::DirectConnection);
               ^~~~~~~
src/mainwindow.cpp:456:2: note: in instantiation of function template specialization 'QObject::connect<void (SQLToolWidget::*)(), (lambda at src/mainwindow.cpp:456:68)>' requested here
        connect(sql_tool_wgt, &SQLToolWidget::s_connectionsUpdateRequest, [&](){ updateConnections(true); });
        ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:273:13: note: candidate function template not viable: no known conversion from 'const typename QtPrivate::FunctionPointer<void (SQLToolWidget::*)()>::Object *' (aka 'const SQLToolWidget *') to 'const QObject *' for 3rd argument
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:318:13: note: candidate function template not viable: no known conversion from 'const typename QtPrivate::FunctionPointer<void (SQLToolWidget::*)()>::Object *' (aka 'const SQLToolWidget *') to 'const QObject *' for 3rd argument
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, const QObject *context, Func2 slot,
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:212:36: note: candidate function not viable: no known conversion from 'const typename QtPrivate::FunctionPointer<void (SQLToolWidget::*)()>::Object *' (aka 'const SQLToolWidget *') to 'const QObject *' for 1st argument
    static QMetaObject::Connection connect(const QObject *sender, const char *signal,
                                   ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:215:36: note: candidate function not viable: no known conversion from 'const typename QtPrivate::FunctionPointer<void (SQLToolWidget::*)()>::Object *' (aka 'const SQLToolWidget *') to 'const QObject *' for 1st argument
    static QMetaObject::Connection connect(const QObject *sender, const QMetaMethod &signal,
                                   ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:232:43: note: candidate template ignored: substitution failure [with Func1 = void (SQLToolWidget::*)(), Func2 = (lambda at src/mainwindow.cpp:456:68)]: no type named 'Object' in 'QtPrivate::FunctionPointer<(lambda at src/mainwindow.cpp:456:68)>'
    static inline QMetaObject::Connection connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal,
                                          ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:219:36: note: candidate function not viable: requires at most 4 arguments, but 5 were provided
    inline QMetaObject::Connection connect(const QObject *sender, const char *signal,
                                   ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:307:13: note: candidate function template not viable: requires 3 arguments, but 5 were provided
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
            ^
/opt/homebrew/lib/QtCore.framework/Headers/qobject.h:264:13: note: candidate function template not viable: requires 3 arguments, but 5 were provided
            connect(const typename QtPrivate::FunctionPointer<Func1>::Object *sender, Func1 signal, Func2 slot)
            ^
8 errors generated.
make[1]: *** [obj/mainwindow.o] Error 1
make: *** [sub-libs-libgui-make_first-ordered] Error 2

```

</details>

**Additional info**
I've successfully built 0.9.4 (qt5) and 1.0.0-alpha (qt6 ver) before using brew libs before so I'm not sure if there's where the problem is.
I was also unable to build it on my older mac using the same steps.

**Comment by rkhaotix** [comment]
I never tried to build pgModeler using brew's Qt, only the official package distributed at qt.io. Anyway,  It seems that you have a problem with your Qt installation... Just a guess. Could you post the entire compile output?

**Comment by Ravcii** [comment]
Sure, Here: https://pastebin.com/PUFxFhGf

From what i can understand it goes smooth up until src/mainwindow.cpp. (Line 859 on pastebin)

I also had some errors like "QWidget has no QAction" or something, but if I recall correctly that's because I was trying to build it using qt5.

**Comment by Ravcii** [comment]
I tried to download QT from qt.io (open source installer for mac) but I got the same errors with both `6.3.1` and `6.2.4`

**Comment by Ravcii** [comment]
I've finally found a solution. The problem was using the wrong import path for libxml includes.
I was using (extra `/libxml` at the end)
```
!defined(XML_INC, var): XML_INC = /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libxml2/libxml
```
but it should be 
```
!defined(XML_INC, var): XML_INC = /Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libxml2
```

I guess it confused me because when i did `ls -a` in `..../libxml2/` it gave me this output
```
/Applications/Xcode-beta.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libxml2 ❯ ls -a                     
.      ..     libxml
```
So I thought there were no headers.

The final build script looks like this. I did remove `macdeployqt` because it was actually break the app for some reason. This is probably not the correct way to build the app according to the Installation guide but it seems to be working.
I also believe this could be built using brew's qt and libxml, but I haven't tested it myself.
```sh
export QT_ROOT=/Users/alex/Qt/6.2.4/macos

$QT_ROOT/bin/qmake -r CONFIG+=release pgmodeler.pro \
    && make \
    && make install 
```

So, to summarize:
1. **libpq** from brew
2.  **libxml** from XCode
3.  QT from qt.io installer

I hope this could help somebody.
